src/org/omegat/core/matching/NearString.java

```diff
@@ -49,37 +49,81 @@
  */
 public class NearString {
     public enum MATCH_SOURCE {
-        MEMORY, TM, FILES
-    };
+        /** From project memory */
+        MEMORY,
+        /** From external TM in project tm/ folder */
+        TM,
+        /** From files */
```

Comment:
/* From source files */
(let this line give a useful information, remember that nobody remembered its meaning...)